src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -1230,6 +1230,16 @@ object LazyList extends SeqFactory[LazyList] {
     at(0)
   }
 
+  override def range[A: Integral](start: A, end: A): LazyList[A] =
+    range(start, end, step = Integral[A].one)
+
+  override def range[A](start: A, end: A, step: A)(implicit ev: Integral[A]): LazyList[A] =
+    newLL(rangeImpl(start, end, step))
```

Comment:
I totally forgot about an important check, sorry :grimacing:

```suggestion
    newLL {
      if (ev.equiv(step, ev.zero)) throw new IllegalArgumentException("step cannot be 0.")
      else rangeImpl(start, end, step)
    }
```